### PR TITLE
More editor tests

### PR DESCRIPTION
**Pull request by hughbe**
Contributes to #817

### CODE REVIEW

**Comment by hughbe** [comment]
Looks like a real bug:
> System.Resources.MissingManifestResourceException : Could not find the resource \"System.Windows.Forms.Design.ShortcutKeysEditor.resources\" among the resources \"System.ComponentModel.Design.CollectionEditor.resources\", \"System.SR.resources\" embedded in the assembly \"System.Windows.Forms.Design.Editors\", nor among the resources in any satellite assemblies for the specified culture. Perhaps the resources were embedded with an incorrect name.

**Comment by RussKie** [comment]
Looks like you're blocked by https://github.com/dotnet/winforms/pull/941

**Comment by RussKie** [comment]
@hughbe I superseded #941 with #1229. I have also cherry-picked this changeset locally and all tests appeared to have passed.
Please rebase on top of #1229.

**Comment by RussKie** [comment]
@hughbe your work is extremely appreciated but we can't take large changesets. They are require a lot of effort on both your and our parts, can be difficult to review and ultimately take a lot of time to process. A risk of a bug/regression rises with the size of a changeset as well.
Could you please break the work into a number of smaller PRs - one for each editor.

Thank you for understanding.

**Comment by hughbe** [comment]
Sure - I'll separate them out :)

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/DockEditor.cs [hunk 1]
```diff
@@ -27,20 +26,17 @@ public override object EditValue(ITypeDescriptorContext context, IServiceProvide
         {
             if (provider != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/DockEditor.cs]: Please invert `if` to reduce nesting

File: src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/DockEditor.cs [hunk 2]
```diff
@@ -27,20 +26,17 @@ public override object EditValue(ITypeDescriptorContext context, IServiceProvide
...
             {
-                IWindowsFormsEditorService edSvc =
-                    (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
-
-                if (edSvc != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Windows/Forms/Design/DockEditor.cs]: ditto, and the next line can be reverted

**Pull request merged**